src/main/java/org/example/jvspringbootfirstbook/controller/OrderController.java

```diff
@@ -0,0 +1,69 @@
+package org.example.jvspringbootfirstbook.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.example.jvspringbootfirstbook.dto.order.OrderDto;
+import org.example.jvspringbootfirstbook.dto.order.OrderItemDto;
+import org.example.jvspringbootfirstbook.dto.order.PlacingOrderRequestDto;
+import org.example.jvspringbootfirstbook.model.User;
+import org.example.jvspringbootfirstbook.service.order.OrderService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+
+    @PostMapping
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_USER')")
+    @Operation(summary = "Make Order",
+            description = "make order for user with your shipping address")
+    public OrderDto makeOrder(@RequestBody @Valid PlacingOrderRequestDto requestDto) {
+        return orderService.makeOrder(requestDto);
+    }
+
+    @GetMapping
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_USER')")
+    @Operation(summary = "get order history",
+            description = "get your all orders")
+    public List<OrderDto> showOrderHistory(Authentication authentication, Pageable pageable) {
+        User user = (User) authentication.getPrincipal();
+        return orderService.getHistory(user, pageable);
+    }
+
+    @GetMapping("/{orderId}/items")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_USER')")
+    @Operation(summary = "get all items",
+            description = "get all items from your order")
+    public List<OrderItemDto> getItemsFromOrder(@PathVariable Long orderId) {
+        return orderService.getFromOrder(orderId);
+    }
+
+    @GetMapping("/{orderId}/items/{id}")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_USER')")
+    @Operation(summary = "get one item",
+            description = "get one item from your order")
+    public OrderItemDto getOneItemFromOrder(@PathVariable Long orderId, @PathVariable Long id) {
+        return orderService.getFromOrder(orderId,id);
```

Comment:
Pls add space before id.
It's always worth to format a file (⌘⌥L (macOS) / Ctrl+Alt+L (Windows/Linux)